sdccc/src/main/java/com/draeger/medical/sdccc/util/junit/XmlReportWriter.java

```diff
@@ -261,7 +262,16 @@ private static void writeFailureOrError(final XMLStreamWriter xmlWriter, final R
                 xmlWriter.writeAttribute("message", err.getMessage());
             }
             xmlWriter.writeAttribute("type", err.getClass().getCanonicalName());
-            xmlWriter.writeCData(ExceptionUtils.getStackTrace(err));
+
+            final var stackTrace = ExceptionUtils.getStackTrace(err);
+            final var stackTraceSplit = stackTrace.split("]]>");
+            final var lastIndex = stackTraceSplit.length - 1;
+            for (final String x :
+                    Arrays.stream(stackTraceSplit).limit(lastIndex).toArray(String[]::new)) {
+                xmlWriter.writeCData(x + "]]");
+                xmlWriter.writeCData(">");
+            }
+            xmlWriter.writeCData(stackTraceSplit[lastIndex]);
```

Comment:
xmlWriter.writeCData() is called a lot in this class. Why do we sanitize only this occurence? Can we be sure that the other Strings do not contain any "]]>"?

Reply:
For most of them it should not be a problem. So far it is only known for failures or errors, because in these cases sometimes the whole status or descriptor is logged.

Reply:
Of course we can also fix the problems as they occur, but thinking proactively usually pays off in the long run.
Does anything speak against extracting your improved CDATA-encoding into a method and applying it whereever a String is written into a CDATA field?

Reply:
No not really, extracted the CDATA handling into a method and used the method where unknown strings are used.